distribution/pull_v2.go

```diff
@@ -922,6 +923,14 @@ func (p *puller) pullSchema2Config(ctx context.Context, dgst digest.Digest) (con
 	return configJSON, nil
 }
 
+type noMatchesErr struct {
+	platform specs.Platform
+}
+
+func (e noMatchesErr) Error() string {
```

Comment:
For a follow-up, we should check what status code this produces at the API level (and if needed, use a suitable errdef